Source/Dafny/Translator.cs

```diff
@@ -12925,50 +12939,7 @@ Bpl.Expr MkIsAlloc(Bpl.Expr x, Type t, Bpl.Expr h)
-                var nm = varNameGen.FreshId(string.Format("#i{0}#", ii));
-                var bv = new Bpl.BoundVariable(tok, new Bpl.TypedIdent(tok, nm, Bpl.Type.Int));
-                bvs.Add(bv);
-                var ie = new Bpl.IdentifierExpr(tok, bv);
-                indices.Add(ie);
-                ante = BplAnd(ante, BplAnd(Bpl.Expr.Le(Bpl.Expr.Literal(0), ie), Bpl.Expr.Lt(ie, etran.TrExpr(tRhs.ArrayDimensions[ii]))));
-              }
-              var sourceType = tRhs.ElementInit.Type.AsArrowType;
-              Contract.Assert(sourceType.Args.Count == tRhs.ArrayDimensions.Count);
-              var args = Concat(
-                Map(Enumerable.Range(0, tRhs.ArrayDimensions.Count), ii => TypeToTy(sourceType.Args[ii])),
-                Cons(TypeToTy(sourceType.Result),
-                Cons(etran.HeapExpr,
-                Cons(etran.TrExpr(tRhs.ElementInit),
-                indices.ConvertAll(idx => (Bpl.Expr)FunctionCall(tok, BuiltinFunction.Box, null, idx))))));
-              // check precond
-              var pre = FunctionCall(tok, Requires(tRhs.ArrayDimensions.Count), Bpl.Type.Bool, args);
-              var q = new Bpl.ForallExpr(tok, bvs, Bpl.Expr.Imp(ante, pre));
-              builder.Add(AssertNS(tok, q, "all array indices must be in the domain of the initialization function"));
-              // Check that the values coming out of the function satisfy any appropriate subset-type constraints
-              var apply = UnboxIfBoxed(FunctionCall(tok, Apply(tRhs.ArrayDimensions.Count), TrType(tRhs.EType), args), tRhs.EType);
-              string msg;
-              var cre = GetSubrangeCheck(apply, sourceType.Result, tRhs.EType, out msg);
-              if (cre != null) {
-                // assert (forall i0,i1,i2,... ::
-                //            0 <= i0 < ... && ... ==> tRhs.ElementInit.requires(i0,i1,i2,...) is Subtype);
-                q = new Bpl.ForallExpr(tok, bvs, Bpl.Expr.Imp(ante, cre));
-                builder.Add(AssertNS(tRhs.ElementInit.tok, q, msg));
-              }
-              // Assume that array elements have initial values according to the given initialization function.  That is:
-              // assume (forall i0,i1,i2,... :: { nw[i0,i1,i2,...] }
-              //            0 <= i0 < ... && ... ==> nw[i0,i1,i2,...] == tRhs.ElementInit.requires(i0,i1,i2,...));
-              var ai = ReadHeap(tok, etran.HeapExpr, nw, GetArrayIndexFieldName(tok, indices));
-              var ai_prime = UnboxIfBoxed(ai, tRhs.EType);
-              var tr = new Bpl.Trigger(tok, true, new List<Bpl.Expr> { ai });
-              q = new Bpl.ForallExpr(tok, bvs, tr, Bpl.Expr.Imp(ante, Bpl.Expr.Eq(ai_prime, apply)));  // TODO: use a more general Equality translation
-              builder.Add(new Bpl.AssumeCmd(tok, q));
+              CheckElementInit(tok, true, tRhs.ArrayDimensions, tRhs.EType, tRhs.ElementInit, nw, builder, etran, new WFOptions());
```

Comment:
Nice to move this bunch of code to a method.  Is the code changed at all?  It's hard to review code movement changes.

Reply:
I think the movement did not change the previous code, only parameterized it.

Is there some way you'd prefer that such refactorings be marked or pointed out in the pull requests / code reviews? I commonly rearrange code to make it better or more usable.

Reply:
Yeah, I don't know a good way.  On other teams we've enforced a policy that code movement goes in separate PRs so we don't need to use brain cycles diffing the two versions in our heads.  That adds some overhead.  Alternatively, you can comment on your own PR saying something like "this code is just moved, no logical changes".  The danger of course is that when moving code you also change it in a way that introduced a bug, and the diff makes the change very difficult to find.